File: GraffiTab-iOS/Classes/ViewControllers/Main/Main/Settings/GeneralSettingsViewController.m
//
//  GeneralSettingsViewController.m
//  GraffiTab-iOS
//
//  Created by Georgi Christov on 25/04/2015.
//  Copyright (c) 2015 GraffiTab. All rights reserved.
//

#import "GeneralSettingsViewController.h"
#import "InfoViewController.h"

@interface GeneralSettingsViewController ()

@end

@implementation GeneralSettingsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)showInfoController:(NSString *)file title:(NSString *)title {
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"MainStoryboard" bundle:nil];
    InfoViewController *vc = [storyboard instantiateViewControllerWithIdentifier:@"InfoViewController"];
    vc.filePath = file;
    vc.title = title;
    
    [self showController:vc];
}

- (void)showController:(UIViewController *)vc {
    [self.navigationController pushViewController:vc animated:YES];
}

@end

File: GraffiTab/Classes/ViewControllers/Home/Settings/GeneralSettingsViewController.swift
//
//  GeneralSettingsViewController.swift
//  GraffiTab
//
//  Created by Georgi Christov on 12/04/2016.
//  Copyright © 2016 GraffiTab. All rights reserved.
//

import UIKit

class GeneralSettingsViewController: BackButtonTableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        
        // Register analytics events.
        AnalyticsUtils.sendScreenEvent(self)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func showInfoViewController(title: String, file: String) {
        let storyboard = UIStoryboard(name: "LoginStoryboard", bundle: NSBundle.mainBundle())
        let infoVC = storyboard.instantiateViewControllerWithIdentifier("WebViewController") as! WebViewController
        infoVC.title = title
        infoVC.filePath = file
        self.navigationController?.pushViewController(infoVC, animated: true)
    }
}
